### ISSUE TITLE
Bug: (fill in)

### ISSUE THREAD

**Issue by joket1999** [eslint/eslint#16858]
### Environment

Node version: v14.21.1
npm version: v6.14.17
Local ESLint version: v8.33.0
Global ESLint version: v8.33.0
Operating System: MacOS 10.15.7

### What parser are you using?

@babel/eslint-parser

### What did you do?

<details>
<summary>Configuration</summary>

```
{
"require-jsdoc": ["warn", {
	"require": {
		"FunctionDeclaration": true,
		"MethodDefinition": true,
		"ClassDeclaration": true,
		"ArrowFunctionExpression": true,
		"FunctionExpression": true
	}
}]
}
```
</details>

```js
function *generator() {
	yield '44';
	yield '55';
}
```

### What did you expect to happen?

Is there a way to disable require-jsdoc checks on generator functiuon

### What actually happened?

eslint: warning
require-jsdoc - Missing JSDoc comment.

### Participation

- [ ] I am willing to submit a pull request for this issue.
